caikit/runtime/service_generation/primitives.py

```diff
@@ -45,52 +45,12 @@ def to_primitive_signature(
     primitives = {}
     log.debug("Building primitive signature for %s", signature)
     for arg, arg_type in signature.items():
-        primitive_arg_type = extract_primitive_type_from_union(
-            primitive_data_model_types, arg_type
-        )
-        if primitive_arg_type:
-            primitives[arg] = primitive_arg_type
+        if is_primitive_type(arg_type, primitive_data_model_types):
```

Comment:
Not sure if the function name `to_primitive_signature` holds true anymore - but don't want to change a lot since other PRs are touching the same stuff too